### ISSUE TITLE
Add support for uintptr encoding/decoding

### ISSUE THREAD

**Issue by anisse** [go-restruct/restruct#57]
Calling Linux ioctls, one might sometime find the need to use packed structs, and restruct's Pack/Unpack is perfect for that so we don't need to manipulate elements with encoding/binary.

It would be nice to add support for uintptr encoding. Right now it returns 0 in sizeof and isn't encoded. It needs special handling to support 32 and 64 bits platforms, but otherwise it should be doable. Are you interested in a PR for this ?

**Comment by jchv** [comment]
With the way restruct works today, its encoding is agnostic; it should basically encode a given structure identically regardless of endianness or processor word size. Maybe we should do *something* about uintptr, but it's definitely tricky: if you force it to always encode as 64 bit, you'll have consistent encoding no matter what, but if you decode from 64 bit to a 32 bit machine, it could possibly lose information.

My inclination is that it'd be nice to support for the logical in-memory use case, though. I'm leaning in this direction: always encode like uint64, BUT return an error if decoding back to uintptr would overflow. WDYT? I'm open to more discussion here, but it's definitely a tricky one!

(Edit: it's also convenient if we can maintain the consistency of encodings across archs and platforms *especially* because I'd like to still eventually do a fast code gen for restruct, and it'd be annoying to have to add arch-specific files in the code gen just to deal with this.)

**Comment by anisse** [comment]
I like your proposal to return an error during uintptr encode/decode on 32 bits, because then you decide that encoding uintptr on 32bits platform is out of scope - almost like it currently is. I don't foresee using a 32 bits platform for my use case.

One thing to look for: hopefully users don't have structs with uintptr right now (ignored), because a new version would break compatibility by adding new data where there was none before. And on 32 bits there would be an error where the function previously worked.

**Comment by jchv** [comment]
True: I think we would need to save such a change for restruct 2. I'm going to try to put some more thought into restruct 2 going into next year as there are quite a lot of improvements and cleanups that could be made.

**Comment by fritzr** [comment]
Whatever solution you implement here should probably also address int and uint - see #58.
